packages/domainproperties/src/components/assay/AssayPropertiesPanel.tsx

```diff
@@ -112,23 +163,50 @@ export class AssayPropertiesPanel extends React.PureComponent<Props, State> {
     };
 
     onValueChange = (id, value) => {
-        this.props.onChange(this.props.model.merge({
+        const { model } = this.props;
+
+        const newModel = model.merge({
             [id.replace(FORM_ID_PREFIX, '')]: value
-        }));
+        }) as AssayProtocolModel;
+
+        const valid = (newModel.hasValidProperties() === true ? true : this.state.validProperties);
+
+        this.setState((state) => (
+                // Only clear validation errors here. New errors found on collapse or submit.
+                {validProperties: valid}),
+            () => {
+                this.props.onChange(newModel);
+            });
+
     };
 
+    getHeaderClasses(): string {
```

Comment:
Same comment as in DomainForm regarding the function names for getHeaderClasses vs getHeaderClass. Also, since this function body and the getHeaderClass one for icons are very similar. Is there any way to share it between the two components? Maybe move them to actions.ts or soemthing along those lines?